pkg/k8s/smi.go

```diff
@@ -0,0 +1,57 @@
+package k8s
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	access "github.com/servicemeshinterface/smi-sdk-go/pkg/apis/access/v1alpha2"
+	specs "github.com/servicemeshinterface/smi-sdk-go/pkg/apis/specs/v1alpha3"
+	split "github.com/servicemeshinterface/smi-sdk-go/pkg/apis/split/v1alpha3"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes"
+)
+
+// CheckSMIVersion checks if the SMI CRDs versions installed matches what Maesh supports.
+func CheckSMIVersion(client kubernetes.Interface, aclEnabled bool) error {
+	serverGroups, err := client.Discovery().ServerGroups()
+	if err != nil {
+		return fmt.Errorf("unable to list kubernetes server groups: %w", err)
+	}
+
+	requiredGroups := []schema.GroupVersion{
+		split.SchemeGroupVersion,
+		specs.SchemeGroupVersion,
+	}
+
+	if aclEnabled {
+		requiredGroups = append(requiredGroups, access.SchemeGroupVersion)
+	}
+
+	var errs []string
+
+	for _, requiredGroup := range requiredGroups {
+		var found bool
+
+		var version string
+
+		for _, group := range serverGroups.Groups {
+			if requiredGroup.Group == group.Name {
+				found = true
```

Comment:
I'm not sure we need the found `bool` here and I think we can add a break.

Reply:
I could definitely live without this variable but the issue was to report every CRD with a mismatched version.